scarb/src/ops/package.rs

```diff
@@ -88,10 +98,48 @@ fn before_package(ws: &Workspace<'_>) -> Result<()> {
     Ok(())
 }
 
-#[tracing::instrument(level = "trace", skip(_opts, ws))]
+#[derive(Serialize)]
+struct GitVcsInfo {
+    sha1: String,
+}
+
+#[derive(Serialize)]
+struct VcsInfo {
+    git: GitVcsInfo,
+    path_in_vcs: String,
+}
+
+fn has_vcs(pkg: &Package) -> bool {
+    gix::discover(pkg.root().to_path_buf()).is_ok()
+}
+
+fn extract_vcs_info(pkg: &Package, opts: &PackageOpts) -> Result<Option<VcsInfo>> {
+    if let Ok(repo) = PackageRepository::open(pkg) {
+        ensure!(
+            opts.allow_dirty || repo.is_clean()?,
+            indoc!(
+                r#"
+                cannot package a repository containing uncommited changes
+                help: to proceed despite this and include the uncommitted changes, pass the `--allow-dirty` flag
+                "#
+            )
```

Comment:
```suggestion
            indoc! {r#"
                cannot package a repository containing uncommited changes
                help: to proceed despite this and include the uncommitted changes, pass the `--allow-dirty` flag
            "#}
```

using `{}` in this context is more idiomatic. one indent less

we use `()` only in the following cases: `formatdoc!("pattern", arg, ...)`